apps/backoffice-v2/src/lib/blocks/hooks/useEntitiesDocuments/useEntitiesDocuments.ts

```diff
@@ -0,0 +1,119 @@
+import { useDocumentsAdapter } from '@/domains/documents/hooks/useDocumentsAdapter/useDocumentsAdapter';
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { selectDirectors } from '@/pages/Entity/selectors/selectDirectors';
+import { TDocument } from '@ballerine/common';
+import { useCallback, useMemo } from 'react';
+
+export type TEntitiesDocumentsFilter = 'all' | 'directors' | 'ubos';
+
+const allOrExact = (filter: TEntitiesDocumentsFilter, exact: TEntitiesDocumentsFilter) =>
+  filter === 'all' || filter === exact;
+
+export const useEntitiesDocuments = (
+  workflow?: TWorkflowById,
```

Comment:
Workflow is a must, right? It's not optional it just can be fetched.
```suggestion
  workflow: TWorkflowById | undefined,
```

Reply:
You can't render hooks conditionally. In some cases workflow recieved from query, which is either Workflow or undefined